Cannot be used in conjunction with github.com/lib/pq

Attempting to use this for a project where I'm hoping to also support postgres as well as cockroach db. However, since this package imports "github.com/cockroachdb/pq" it's impossible to also register the "github.com/lib/pq" postgres driver. The registration panics because they both want to be the "postgres" driver.

```
panic: sql: Register called twice for driver postgres

goroutine 1 [running]:
panic(0x714d60, 0xc420010d50)
	/usr/local/go/src/runtime/panic.go:500 +0x1a1
database/sql.Register(0x76b2ba, 0x8, 0xabda60, 0xaefe88)
	/usr/local/go/src/database/sql/sql.go:45 +0x1b2
github.com/lib/pq.init.1()
	/home/eric/src/github.com/lib/pq/conn.go:44 +0x5c
github.com/lib/pq.init()
	/home/eric/src/github.com/lib/pq/user_posix.go:25 +0x62f
github.com/coreos/dex/storage/sql.init()
```

What's the reason for using the fork? Is there a plan to get any required patches back into the upstream repo?

--- a/crdb/tx.go
+++ b/crdb/tx.go
@@ -21,7 +21,7 @@ package crdb
 import (
 	"database/sql"
 
-	"github.com/cockroachdb/pq"
+	"github.com/lib/pq"
 )
 
 // AmbiguousCommitError represents an error that left a transaction in an

--- a/crdb/tx_test.go
+++ b/crdb/tx_test.go
@@ -50,14 +50,17 @@ INSERT INTO d.t (acct, balance) VALUES (1, 100), (2, 100);
 		if err != nil {
 			return
 		}
-		for i, bal := range []*int{&bal1, &bal2} {
-			if !rows.Next() {
-				err = fmt.Errorf("expected two balances; got %d", i)
+		defer rows.Close()
+		balances := []*int{&bal1, &bal2}
+		i := 0
+		for ; rows.Next(); i += 1 {
+			if err = rows.Scan(balances[i]); err != nil {
 				return
 			}
-			if err = rows.Scan(bal); err != nil {
-				return
-			}
+		}
+		if i != 2 {
+			err = fmt.Errorf("expected two balances; got %d", i)
+			return
 		}
 		return
 	}

